docs/docs/typescript/strict-mode.md

```diff
@@ -0,0 +1,239 @@
+```
+
+## Manual tweaks to generated code
+
+Now that you're in strict mode, there are some changes you need to make to get rid of those pesky red underlines!
+
+### Service functions & tests
+
+By default, Redwood's generators assume that Service functions don't require any parameters. For example, [in the tutorial](../tutorial/chapter2/side-quest.md), the `posts` Service function generated by the scaffold command doesn't specify any parameters:
+
+```ts title="src/services/posts/posts.test.ts"
+// highlight-next-line
+export const posts: QueryResolvers['posts'] = () => {
+  return db.post.findMany()
+}
+```
+
+While this is true, in the strictest sense, all GraphQL resolvers (and most of your Service functions _do_ end up becoming GraphQL resolvers) take some sort of input, so you'll need to modify the calls to your Service functions that have been typed with `QueryResolvers` or `MutationResolvers`:
+
+```ts title="src/services/posts/posts.test.ts"
+describe('posts', () => {
+  scenario('returns all posts', async (scenario: StandardScenario) => {
+    const result = await posts() // 🛑 error
+    const result = await posts({}) // ✅
+
+    expect(result.length).toEqual(Object.keys(scenario.post).length)
+  })
+})
+```
+
+:::info Why this happens in strict mode
+
+In strict mode, Redwood generates different types for `QueryResolvers` and `MutationResolvers`, where the first argument—`args`—is no longer optional.
+
+We realize that this isn't a perfect solution, but the tension comes from the fact that we can't define different types for when a Service function is called from inside GraphQL (as a resolver) versus when it's called from anywhere else.
+
+:::
+
+### Returning Prisma's `findUnique` operation in Services
+
+In strict mode, TypeScript becomes a lot more pedantic about null checks. One place where you'll encounter this in particular is when returning Prisma's `findUnique` operation in Service functions.
+
+The tension here is that Prisma returns promises in the form of `Promise<Model | null>`, but the resolver types expect it in the form of `Promise<Model> | Promise<null>`. At runtime, this has no effect. But the TS compiler needs to be told that it's okay:
+
+```ts
+export const post: QueryResolvers['post'] = ({ id }) => {
+  return db.post.findUnique({
+    where: { id },
+    // highlight-next-line
+  }) as Promise<Post> | Promise<null>
+}
+```
+
+### `null` and `undefined` in Services
+
+One of the challenges in the GraphQL-Prisma world is the difference in the way they treats optionals:
+
+- for GraphQL, optional fields can be `null`
+- but For Prisma, `null` is a value, and `undefined` means "do nothing"
+
+This is covered in detail in [Prisma's docs](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined), which we strongly recommend reading.
+But the gist of it is that, for Prisma's create and update operations, you have to make sure `null`s are converted to `undefined` from your GraphQL mutation inputs.
```

Comment:
The way I read this it sounds as if this would be needed even if we're not in strict mode. Should we call this out somewhere in our "regular" docs?

Reply:
Do you know of a good place to put this, so that JS and non-strict-mode TS devs can find this info too? Maybe somewhere in our general "Service" docs? Or in our GraphQL docs?

Reply:
Nope - I think we need to explore our options more on this. Perhaps making dnull part of the template, but really is out of scope for this PR.

Reply:
> Nope - I think we need to explore our options more on this. Perhaps making dnull part of the template, but really is out of scope for this PR.

Sure. Should I write up an issue, or do you have a TODO in your own notes for this somewhere already?

Reply:
Created an issue https://github.com/redwoodjs/redwood/issues/6016